controllers/nim_account_controller.go

```diff
@@ -422,6 +425,47 @@ func (r *NimAccountReconciler) cleanupResources(ctx context.Context, account *v1
 
 }
 
+func (r *NimAccountReconciler) TechPreviewCleanups() manager.RunnableFunc {
```

Comment:
AFAIU this function should also cleanup up the api key secret if present from previous TP release. The other resources are something that is managed by operator so it can remain as you feel.